torchtext/prototype/t5/modules.py

```diff
@@ -53,8 +54,244 @@ def __init__(
         self.v_proj_weight = nn.Parameter(torch.empty((embed_dim, self.vdim), **factory_kwargs))
         self.register_parameter("in_proj_weight", None)
 
-    def forward():
-        pass
+    def forward(
+        self,
+        query: Tensor,
+        key: Tensor,
+        value: Tensor,
+        key_padding_mask: Optional[Tensor] = None,
+        need_weights: bool = True,
+        attn_mask: Optional[Tensor] = None,
+        average_attn_weights: bool = False,
+        compute_relative_attention_bias=False,
+        relative_attention_num_buckets=32,
+        relative_attention_max_distance=128,
+        relative_attention_bias: Optional[Tensor] = None,
+        position_bias: Optional[Tensor] = None,
+    ) -> Tuple[Tensor, Optional[Tensor]]:
```

Comment:
Looks like the return type doesn't match actual return. There are three variable being returned whereas annotation accounts for only 2?